nav2_amcl/CMakeLists.txt

```diff
@@ -20,72 +20,111 @@ find_package(pluginlib REQUIRED)
 
 nav2_package()
 
-include_directories(
-  include
-)
-
 include(CheckSymbolExists)
 check_symbol_exists(drand48 stdlib.h HAVE_DRAND48)
 
-add_subdirectory(src/pf)
-add_subdirectory(src/map)
-add_subdirectory(src/motion_model)
-add_subdirectory(src/sensors)
+if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wno-gnu-folding-constant)
```

Comment:
Can you explain this for AMCL specifically?

Reply:
Yeah, this is a good question.  The honest truth is that for reasons I don't totally understand, having the targets in the subdirectory doesn't produce `nav2_amcl` targets that "worked".  In particular, the downstream package linking against it (`nav2_system_tests`) would fail to properly find the target.  Moving this all together into the main CMakeLists.txt seemed to make it work.  It bears more investigation into why that is happening.